### PR TITLE
solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**
[DEMO LINK](https://nikitakarbazin.github.io/react_movies-list-filter/)

### CODE REVIEW

**Review by averstiuk — APPROVED** [review]
I like your decision)))

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [filterName, setFilterName] = useState('');
+
+  const visibleMovies = moviesFromServer
+    .filter(movie => movie.title.toLowerCase()
+      .includes(filterName.toLowerCase())
+    || movie.description.toLowerCase()
+      .includes(filterName.toLowerCase()));
```
Comment (jstmpelowycz) [inline src/App.tsx:13]: decompose this logic

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (NikitaKarbazin) [reply src/App.tsx:13]: Create another variables? Don't understand what exactly I need to decompose

Comment (jstmpelowycz) [reply src/App.tsx:13]: creating a var is one of the possible ways to make it more readable and maintainable
why not to create some function-helper? as an example
```
const containsString = (...) => ...

const lowercaseQuery = ...

const visibleMovies = movies.fromServer.filter(({ ... }) => {
  const hasTitle = ...
  const hasDescription = ...

  return hasTitle || hasDescription;
});
```

Comment (jstmpelowycz) [reply src/App.tsx:13]: is it good?

**Review by jstmpelowycz — APPROVED** [review]
awesome 🔥

**Pull request closed without merging**